Building libpng Under Clang Fails

`libpng` build is looking for `gcc` compiler. I assume under Clang it should be looking for `clang`?

```
==> Starting build()...
configure: loading site script /etc/config.site
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for x86_64-w64-mingw32-gcc... gcc
checking whether the C compiler works... no
configure: error: in `/home/Jordan/Development/MSYS/MINGW-packages/mingw-w64-libpng/src/build-x86_64-w64-mingw32':
configure: error: C compiler cannot create executables
See `config.log' for more details
==> ERROR: A failure occurred in build().
    Aborting...
```

```
$ echo $MSYSTEM
CLANG64

$ which gcc
which: no gcc in (/clang64/bin:/usr/local/bin:/usr/bin:/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)

$ which clang
/clang64/bin/clang
```

**Edit:** The same error occurs with `bzip2`, `libiconv`, `readline`, & `termcap` packages.

> If `makepkg-mingw` is not possible, then you should pass `--config /etc/makepkg_mingw.conf` to `makepkg`.

Thanks. Problem persists though with a clean `src` dir.

```
checking for x86_64-w64-mingw32-gcc... clang
checking whether the C compiler works... no
configure: error: in `MINGW-packages/mingw-w64-libpng/src/build-x86_64-w64-mingw32':
configure: error: C compiler cannot create executables
See `config.log' for more details
==> ERROR: A failure occurred in build().
    Aborting...
```

But it's looking for the right compiler at least.

CI fails in a different location:

```
  /clang64/bin/gcc -c -I/C/_/mingw-w64-libpng/src/libpng-1.6.37 -I/C/_/mingw-w64-libpng/src/build-x86_64-w64-mingw32  -march=x86-64 -mtune=generic -O2 -pipe png2pnm.c
  make: /clang64/bin/gcc: No such file or directory
  make: *** [Makefile:37: png2pnm.o] Error 127
  ==> ERROR: A failure occurred in build().
      Aborting...
```

I'm guessing that is what @Biswa96 referred to:
> But the second step failed in libpng due to gcc being added in CC variable.

--- a/mingw-w64-libpng/PKGBUILD
+++ b/mingw-w64-libpng/PKGBUILD
@@ -31,9 +31,9 @@ prepare() {
 }
 
 build() {
-  [[ -d "${srcdir}/build-${MINGW_CHOST}" ]] && rm -rf "${srcdir}/build-${MINGW_CHOST}"
-  mkdir -p "${srcdir}/build-${MINGW_CHOST}"
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  [[ -d "${srcdir}/build-${MSYSTEM}" ]] && rm -rf "${srcdir}/build-${MSYSTEM}"
+  mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"
+
   ../"${_realname}-${pkgver}"/configure \
     --build=${MINGW_CHOST} \
     --host=${MINGW_CHOST} \
@@ -41,7 +41,9 @@ build() {
     --prefix=${MINGW_PREFIX} \
     --enable-shared --enable-static \
     as_ln_s="cp -pR"
+
   make
+
   msg2 "Build contributed programs"
   mkdir -p ".libs/contrib/pngminus" && cd ".libs/contrib/pngminus"
   cp -r "${srcdir}/${_realname}-${pkgver}/contrib/pngminus"/* .
@@ -51,8 +53,8 @@ build() {
 # we have to link to libpng16.dll.a or libpng16.a if building 
 # statically. When building the static programs, we might also
 # want to build with the static version of zlib.
-  make PNGINC="-I${srcdir}/${_realname}-${pkgver} -I${srcdir}/build-${MINGW_CHOST}" \
-    CC="${MINGW_PREFIX}/bin/gcc" \
+  make PNGINC="-I${srcdir}/${_realname}-${pkgver} -I${srcdir}/build-${MSYSTEM}" \
+    CC="${MINGW_PREFIX}/bin/cc" \
     PNGLIB_SHARED="-L../.. -lpng16" \
     PNGLIB_STATIC="../../libpng16.a" \
     ZLIB_STATIC="${MINGW_PREFIX}/lib/libz.a" \
@@ -62,13 +64,13 @@ build() {
 }
 
 check() {
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  cd "${srcdir}/build-${MSYSTEM}"
 
   make check
 }
 
 package () {
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  cd "${srcdir}/build-${MSYSTEM}"
   make install DESTDIR="${pkgdir}"
 
   install -D -m644 "${srcdir}/libpng-${pkgver}/LICENSE" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/LICENSE"

